File: sphere.py
import math

class Sphere:
    def __init__(self, ndim=2, radius=1.0):
        self._ndim = ndim
        self._radius = radius

    @property
    def ndim(self):
        return self._ndim

    @ndim.setter
    def ndim(self, value):
        self._ndim = value

    @property
    def radius(self):
        return self._radius

    @radius.setter
    def radius(self, value):
        self._radius = value

    def volume(self):
        return (self._radius ** self._ndim) * (2 * math.pi / 2) ** (self._ndim / 2) / math.gamma(self._ndim / 2 + 1)

    def print_info(self):
        print(f"Dimensions: {self._ndim}, Radius: {self._radius}, Volume: {self.volume()}")

File: Sphere.cpp
#include <iostream>
#include <cmath>
#include "Sphere.hpp" // Include the updated header file

// Constructor definition
Sphere::Sphere(int ndim, double radius) : ndim(ndim), radius(radius) {}

// Getter for ndim
int Sphere::Ndim() { return ndim; }

// Getter for radius
double Sphere::Radius() { return radius; }

// Setter for ndim
void Sphere::Ndim(int ndim) { this->ndim = ndim; }

// Setter for radius
void Sphere::Radius(double radius) { this->radius = radius; }

// Calculate the volume of the n-dimensional sphere
double Sphere::Volume() {
    // Volume formula: (π^(n/2) * R^n) / Γ(n/2 + 1)
    // Note: std::tgamma is used to calculate the Gamma function
    return std::pow(radius, ndim) * std::pow(M_PI, ndim / 2.0) / std::tgamma(ndim / 2.0 + 1);
}

// Print details about the sphere
void Sphere::Print() {
    std::cout << "Dimensions: " << ndim << ", Radius: " << radius << ", Volume: " << Volume() << std::endl;
}
